File: 15-3sum/15-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        HashSet<List<Integer>> result = new HashSet<>();
        for(int i=0;i<nums.length;i++){
            HashMap<Integer, Integer> hashmap = new HashMap<>();
            for(int j=i+1;j<nums.length;j++){
                int sum = -(nums[i]+nums[j]);
                if(hashmap.containsKey(sum)){
                    List<Integer> w = new ArrayList<Integer>();
                    w.addAll(Arrays.asList(sum,nums[i],nums[j]));
                    Collections.sort(w);
                    result.add(w);
                }
                hashmap.put(nums[j],j);
            }
        }
        return new ArrayList<>(result);
    }
}

File: 15-3sum/15-3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result = []
        for i in range(len(nums)):
            d=dict()
            
            for j in range(i+1,len(nums)):
                b = -(nums[i]+nums[j])
                if nums[j] in d:
                    a=[nums[i],nums[j],b]
                    a.sort()
                    if a not in result:
                        result.append(a)
                else:
                    d[b]=nums[j]
        return result
                    
       